Ical.Net/Evaluation/Evaluator.cs

```diff
@@ -43,7 +43,7 @@ protected void IncrementDate(ref CalDateTime dt, RecurrencePattern pattern, int
                 case FrequencyType.Yearly:
                     // When a rule uses BYWEEKNO, recurrence enumeration
                     // is based on week numbers relative to the year.
-                    // So we preserve January 1st when adding years; otherwise, preserve month/day.
+                    // So we preserve the weekday when using BYWEEKNO and preserve month/day otherwise.
                     dt = (pattern.ByWeekNo.Count != 0)
                         ? old.AddDays(-old.DayOfYear + 1).AddYears(interval)
                         : old.AddYears(interval);
```

Comment:
I don't fully understand this change (from #513) though, particularly the non-weekno case. It seems to be somewhat dangerous to just add an interval to the year, especially on Feb 29, as it might not exist in the next interval. Actually, when adding 1y to 2024-02-29, we end up in 2025-02-28. However, when reaching 2028, we still stay on the 28th, even though the year has a 29th. I didn't find any actual issue this is causing, but obviously the monthday doesn't matter. If so, I'd rather recommend setting it to a fixed value (Jan 1 or 1st of the given moth) rather than maintaining a 'wrong' month day. But then, do we need this distinction at all? We actually don't have a test that convers this distinction, i.e. when setting it always to Jan 1st, all tests are still green.